floris/heterogeneous_map.py

```diff
@@ -50,20 +53,40 @@ def __init__(
         if not isinstance(speed_multipliers, (list, np.ndarray)):
             raise TypeError("speed_multipliers must be a numpy array or list")
 
+        # If z is provided, check that it is a list or numpy array
+        if z is not None:
+            if not isinstance(z, (list, np.ndarray)):
```

Comment:
Again, I don't think the nested `if` is necessary here... I would go for 

`if z is not None and not isinstance():`

Reply:
Done!